gitbutler-ui/src/lib/components/BranchLanePopupMenu.svelte

```diff
@@ -47,8 +47,16 @@
 
 			<ContextMenuItem
 				label="Delete"
-				on:click={() => {
-					deleteBranchModal.show(branch);
+				on:click={async () => {
+					if (
+						branch.name.toLowerCase().includes('virtual branch') &&
+						commits.length == 0 &&
+						branch.files?.length == 0
```

Comment:
Not 100% sure whether this is elegant way of detecting whether virtual branch has been updated or not. Adding property something like `touched` might be an option.

Reply:
No this looks decent, thank you!